### PR TITLE
news: GRASS GIS - GSoC 2024 students and projects

### PR DESCRIPTION
**Pull request by veroandreo**
News item about the GSoC projects for GRASS GIS. I set the date to yesterday to be able to render, but we can decide to publish it only later on and change date accordingly. @petrasovaa I put you as author since you are leading GRASS-GSoC issues :)

image

### CODE REVIEW

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+---
+title: "GRASS GIS in Google Summer of Code 2024"
+date: 2024-05-07T14:12:42-05:00
+layout: "news"
+author: Anna Petrasova and the GRASS Development Team
```
Comment (petrasovaa) [inline content/news/2024_05_27_GSoC_2024_students.md:5]: Add yourself there, no?

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+---
+
+### Four Google Summer of Code students will be contributing to GRASS GIS this summer!
```
Comment (petrasovaa) [inline content/news/2024_05_27_GSoC_2024_students.md:8]: Shorter will look better I think
```suggestion
### Four GSoC students will be contributing to GRASS GIS this summer!
```

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+---
+title: "GRASS GIS in Google Summer of Code 2024"
```
Comment (petrasovaa) [inline content/news/2024_05_27_GSoC_2024_students.md:2]:
```suggestion
title: "Google Summer of Code 2024"
```

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+### Four Google Summer of Code students will be contributing to GRASS GIS this summer!
+<a href="https://summerofcode.withgoogle.com/programs/2024/organizations/osgeo-open-source-geospatial-foundation">
+  <img src="https://summerofcode.withgoogle.com/assets/media/logo-sun.svg"
+   alt="GSoC logo"
+   title="GSoC logo"
+   width="20%" style="float:right;padding-left:15px;padding-top:10px">
+</a>
+
+We are thrilled to announce that the 
+**GRASS GIS project got four Google Summer of Code students this year!**
```
Comment (petrasovaa) [inline content/news/2024_05_27_GSoC_2024_students.md:17]:
```suggestion
**GRASS GIS project got four Google Summer of Code (GSoC) students this year!** 
```

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+<a href="https://summerofcode.withgoogle.com/programs/2024/organizations/osgeo-open-source-geospatial-foundation">
+  <img src="https://summerofcode.withgoogle.com/assets/media/logo-sun.svg"
+   alt="GSoC logo"
+   title="GSoC logo"
+   width="20%" style="float:right;padding-left:15px;padding-top:10px">
+</a>
+
+We are thrilled to announce that the 
+**GRASS GIS project got four Google Summer of Code students this year!** 
+They will be working together with their mentors in enhancing
```
Comment (petrasovaa) [inline content/news/2024_05_27_GSoC_2024_students.md:18]:
```suggestion
They will be working together with their mentors on enhancing 
```

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+This project is aimed at parallelizing geospatial tools/algorithms to enhance 
+GRASS GIS capabilities to utilize and analyze large-scale geospatial and remote 
+sensing data. 
+While GRASS GIS already has several tools that offer the option to be run in 
+parallel, and some other were parallelized during GSoC 2021, in this
+opportunity, the targets are:  
+*r.texture, r.proj, r.mapcalc, r.horizon, r.viewshed, v.to.rast, r.to.vect*. 
+Parallelization should significantly accelerate their processing capabilities 
+and bring immediate performance gains in workflows and tools making use of them.
```
Comment (petrasovaa) [inline content/news/2024_05_27_GSoC_2024_students.md:37]:
```suggestion
The goal of this project is to parallelize geospatial tools and algorithms
within GRASS GIS, enhancing its capabilities to process and analyze large-scale
geospatial and remote sensing data efficiently.
While GRASS GIS already includes several tools capable of parallel execution,
and some were parallelized during GSoC 2021, this project aims to
parallelize additional tools to improve their processing speed. Potential targets
include tools like *r.texture, r.proj, r.mapcalc, r.horizon, r.viewshed,
v.to.rast,* and *r.to.vect*. This effort will bring immediate performance
gains to workflows and applications utilizing these tools.
```

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+The python package [`grass.jupyter`](https://grass.osgeo.org/grass83/manuals/libpython/grass.jupyter.html) 
+was developed during GSoC 2021 to simplify running GRASS GIS from Jupyter Notebooks
+and displaying data. 
+This project will focus on enhancing the user experience by adding features such
+as increased interactivity of displayed data to enable query of raster and vector 
+attribute, draw a box to set the computational region or draw simple geometries to 
+add as GRASS vector maps; use of custom projections with `InteractiveMap` class to
+speed up visualization of large maps; and parallelization to speed up rendering of 
+`SeriesMap` and `TimeseriesMap` classes.
```
Comment (petrasovaa) [inline content/news/2024_05_27_GSoC_2024_students.md:53]:
```suggestion
This project aims to enhance the user experience by incorporating new features.
These enhancements will include increased interactivity for displayed data, enabling
the querying of raster and vector attributes, drawing a box to set the computational
region, and creating simple geometries to add as GRASS vector maps. The `InteractiveMap`
class will support custom projections to accelerate the visualization of large maps,
while parallelization will be implemented to expedite rendering in the `SeriesMap`
and `TimeseriesMap` classes.
```

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+Most GRASS GIS tools that output text have custom human readable outputs.
+However, they could benefit from storing their output in a portable and commonly
+used data format like [JSON](https://www.json.org/json-en.html). 
+This project will add JSON as an output format to various existing tools, as well as
+Python test cases to verify the output works as intended. 
+An option to specify the desired output format (plain or JSON) will 
+also be added and the layout of the JSON format will be optimized towards an 
+easy ingestion with Pandas in order to streamline the integration into Python 
+workflows.
```
Comment (petrasovaa) [inline content/news/2024_05_27_GSoC_2024_students.md:75]:
```suggestion
Most GRASS GIS tools currently produce custom human-readable outputs in text
format, but they could greatly benefit from adopting a more portable and widely
recognized data format such as [JSON](https://www.json.org/json-en.html).
This project aims to expand the capabilities of these tools by adding JSON as
an alternative output format.
The layout of the JSON format will be optimized specifically for easy
integration with Pandas, a popular Python library for data analysis, thus
facilitating seamless integration into Python workflows.
This initiative will enhance the versatility and interoperability of GRASS GIS
tools within modern data science environments.
```

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+GRASS GIS has currently various modules to download satellite data: 
+*i.sentinel.download, i.landsat.download, i.modis.download*. However, they all 
+use different libraries, and some of them are no longer maintained. 
+This project will focus on implementing the use of 
+[EODAG](https://eodag.readthedocs.io/en/stable/) 
+as back-end for those tools and thus, decreasing the maintenance burden 
+needed to keep up with all the different APIs updates. 
+Furthermore, since EODAG provides a generalized API as an abstraction of 
+various data providers, there is the potential to significantly extend the 
+currently supported imagery data in GRASS GIS.
```
Comment (petrasovaa) [inline content/news/2024_05_27_GSoC_2024_students.md:92]:
```suggestion
GRASS GIS currently offers several modules for downloading satellite data:
*i.sentinel.download, i.landsat.download, i.modis.download*. However, these
modules use different libraries, and some are no longer maintained. This project
aims to implement the use of [EODAG](https://eodag.readthedocs.io/en/stable/) as
the backend for these tools, reducing the maintenance burden associated with
managing multiple API updates. Additionally, leveraging EODAG's generalized API,
which abstracts various data providers, has the potential to significantly expand
the supported imagery data within GRASS GIS.

```

File: static/images/news/gsoc_2024_students_and_mentors.png
Comment (petrasovaa) [inline static/images/news/gsoc_2024_students_and_mentors.png:1]: Good you took screenshot! I haven't thought about that...

**Review by petrasovaa — APPROVED** [review]
I tried chatgpt on the project descriptions, it's not that much different, but some sentences flow better.

I would still wait with merging this.

File: content/news/2024_05_27_GSoC_2024_students.md (hunk 1 above)
Comment (veroandreo) [inline content/news/2024_05_27_GSoC_2024_students.md:5]:
```suggestion
author: Anna Petrasova, Veronica Andreo and the GRASS Development Team
```

**Comment by veroandreo** [comment]
> I tried chatgpt on the project descriptions, it's not that much different, but some sentences flow better.
> 
> I would still wait with merging this.

It reads better, thanks! My plan was to publish it on May 27 that GSoC officially starts, or a couple of days before that, what do you think @petrasovaa ?

File: content/news/2024_05_27_GSoC_2024_students.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
+---
+title: "Google Summer of Code 2024"
+date: 2024-05-07T14:12:42-05:00
```
Comment (neteler) [inline content/news/2024_05_27_GSoC_2024_students.md:3]:
```suggestion
date: 2024-05-27T14:12:42-05:00
```

Start of GSoC

**Pull request merged**